apis/core/v1alpha2/featuregate_webhook.go

```diff
@@ -0,0 +1,318 @@
+			fmt.Sprintf("cannot toggle features %v as the the stability level of the features indicate that it"+
+				"should not be activated in production environments. To activate the feature, you must agree to "+
+				"permanently void all support guarantees for this environment by setting "+
+				"featureRef.permanentlyVoidAllSupportGuarantees to true", featuresThatVoidWarranty)))
+	}
+
+	if len(immutableFeatures) > 0 {
+		allErrors = append(allErrors, field.Invalid(field.NewPath("spec").Child("features"), r.Spec.Features,
+			fmt.Sprintf("cannot toggle immutable features: %v", immutableFeatures)))
+	}
+
+	return allErrors
+}
+
+// computeFeaturesThatVoidSupportWarranty computes and returns features which voids the support warranty
+func computeFeaturesThatVoidSupportWarranty(spec FeatureGateSpec, features *FeatureList) []string {
+	invalidFeatures := sets.String{}
+	for _, featureRef := range spec.Features {
+		stabilityLevel, found := getFeatureStabilityLevel(features, featureRef.Name)
+		if !found {
+			// Feature doesn't exist and is validated in validateFeatureExistence method
+			continue
+		}
+		policy := GetPolicyForStabilityLevel(stabilityLevel)
+		if policy.VoidsWarranty && !featureRef.PermanentlyVoidAllSupportGuarantees && policy.DefaultActivation != featureRef.Activate {
+			invalidFeatures.Insert(featureRef.Name)
+		}
+	}
+	return invalidFeatures.List()
+}
+
+// computeImmutableFeatures computes and returns features which are immutable
+func computeImmutableFeatures(spec FeatureGateSpec, features *FeatureList) []string {
+	invalidFeatures := sets.String{}
+	for _, featureRef := range spec.Features {
+		stabilityLevel, found := getFeatureStabilityLevel(features, featureRef.Name)
+		if !found {
+			// Feature doesn't exist and is validated in validateFeatureExistence method
+			continue
+		}
+		policy := GetPolicyForStabilityLevel(stabilityLevel)
+		if policy.Immutable && policy.DefaultActivation != featureRef.Activate {
+			invalidFeatures.Insert(featureRef.Name)
+		}
+	}
+	return invalidFeatures.List()
+}
+
+// validateStabilityPolicyOverride determines if permanentlyVoidAllSupportGuarantees field in featureReference in
+// featuregate spec is set to false after setting it to true initially
+func (r *FeatureGate) validateStabilityPolicyOverride(oldObject *FeatureGate) field.ErrorList {
+	var allErrors field.ErrorList
+
+	invalidFeatures := computeInvalidStabilityPolicyOverridedFeatures(r.Spec, oldObject)
+	if len(invalidFeatures) > 0 {
+		allErrors = append(allErrors, field.Invalid(field.NewPath("spec").Child("features"), r.Spec.Features,
+			fmt.Sprintf("cannot toggle features due to policy violation: %v."+
+				"Set feature.skipStabilityPolicyValidation to true to override policy."+
+				"Once set to true, cannot be set back to false.", invalidFeatures)))
+	}
+	return allErrors
+}
+
+// computeInvalidStabilityPolicyOverridedFeatures computes and returns features that have overrided the stability policy override
```

Comment:
```suggestion
// computeInvalidStabilityPolicyOverridedFeatures computes and returns features
// that have overridden the stability policy.
```

Reply:
I see you are much more comfortable with longer line lengths (maybe due to your huge monitor?). I try to stick to 80 characters, but make them longer if I don't break anything up that would be better together. I wish we had a line-length number/range for TF, but we don't, so go ahead an make my above suggestion a longer line. I like to follow the guidelines at https://github.com/golang/go/wiki/CodeReviewComments#line-length

Reply:
restricting the line length to 80 characters is too constraining and outdated, I prefer to limit the line length to 120 characters as we have wide monitors these days and 120 characters doesn't seem constraining. I do see a few lines in my code that exceeded 120 character length, will fix them.

Reply:
Sounds good. I don't like 120 as I think it's too long, but as I said, we have no guidelines, so 120 is all right.